edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -228,6 +228,20 @@ public Object doInHibernate(final Session session) throws HibernateException {
 		}
 	}
 
+	public org.sakaiproject.service.gradebook.shared.Assignment getExternalAssignment(final String gradebookUid, final String externalId)
```

Comment:
You probably want to use the GradebookExternalAssessmentService when dealing with external assessments. As you have witnessed its strange the way these services are separated!

Reply:
I think I'd rather just leave it in there. The returned type is different to the existing (private) external service method and the conversion method is in gradebookservice. It's really messy.